lib/yeti_exception/error.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module YetiException
+
+  # Enhanced exception class, with some features added beyond those in
+  # StandardError:
+  #
+  # - The class in which the exception was raised.  This can be useful for
+  #   logging, since the backtrace is much more verbose.
+  #
+  # - A details hash, containing arbitrary key/value pairs related to the
+  #   exception.
+  #
+  # - A transient flag, defaulting to true, indicating whether the exception
+  #   should be considered transient or final.  This can be used when rescuing
+  #   the exception to determine whether the operation should be retried.
+  #
+  # - An HTTP status, defaulting to 500.  This can be used to generate the
+  #   response payload if the exception is rescued as part of an HTTP request.
+  #
+  # - An optional static message, definable by subclasses, to be included
+  #   automatically in the details hash.  This removes the need to pass this
+  #   message in explicitly during initialization.
+
+  class Error < StandardError
+
+    # @return [Class] Class in which the exception occurred
+    attr_reader :klass
+    # @return [Hash] Arbitrary details about the exception
+    attr_reader :details
+    # @return [TrueClass, FalseClass] Whether the error is transient
+    attr_reader :transient
+    # @return [Integer] Corresponding HTTP status
+    attr_reader :http_status
+
```

Comment:
Just a thought: Might we want to add first class support for a `user_id` on yesware's errors?

Reply:
I wanted to keep this as general as possible.  There could be asynchronous jobs that don't operate in the context of a single user.

Reply:
That same argument would apply to `http_status`, which would never apply in the contezt of an asynchronous job.

Reply:
My view of this is a general exception framework for an app that has both synchronous HTTP traffic and asynchronous retryable workers.  The fact that those actions are performed in the context of or on behalf of a user seems to be less of a generalization.

Reply:
I don't necessarily agree with that, but I do suppose that a `user_id` belongs more at the logging level than it does at the exception level. 👍